macros/src/types/mod.rs

```diff
@@ -46,3 +51,122 @@ fn type_def(attr: &StructAttr, ident: &Ident, fields: &Fields) -> Result<Derived
         Fields::Unit => unit::null(attr, &name),
     }
 }
+
+pub(super) fn type_as_infer(type_as: &Type, original_type: &Type) -> Type {
```

Comment:
This is perfectly fine, but just an idea: Would changing the signature to `type_as_infer(type_as: &Type, original: &mut Type) -> ()` make the impl cleaner? I think that might get rid of the `parse_quote!` calls.

Reply:
I've put something together in #305 to highlight what I had in mind. I've put the `&mut` on the wrong argument in the snippet above, i think that caused some confusion. Sorry about that.

Reply:
> I've put something together in https://github.com/Aleph-Alpha/ts-rs/pull/305 to highlight what I had in mind.

Oh, that makes a lot more sense! I'll merge that in real quick